service/header/rpc.go

```diff
@@ -0,0 +1,114 @@
+package header
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/celestiaorg/celestia-node/node/rpc"
+)
+
+const (
+	headerByHeightEndpoint = "/header"
+	dahByHeightEndpoint    = "/dah"
+)
+
+var (
+	rpcHeightKey = "height"
+)
+
+type dahResponse struct {
+	DAH string `json:"dah"`
+}
+
+func (s *Service) RegisterEndpoints(rpc *rpc.Server) {
+	rpc.RegisterHandlerFunc(fmt.Sprintf("%s/{%s}", headerByHeightEndpoint, rpcHeightKey), s.handleHeaderRequest,
+		http.MethodGet)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("%s/{%s}", dahByHeightEndpoint, rpcHeightKey), s.handleRootRequest,
+		http.MethodGet)
+}
+
+func (s *Service) handleHeaderRequest(w http.ResponseWriter, r *http.Request) {
+	header, err := s.performGetHeaderRequest(w, r, headerByHeightEndpoint)
+	if err != nil {
+		// return here as we've already logged and written the error
+		return
+	}
+	// marshal and write response
+	resp, err := json.Marshal(header)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Errorw("serving request", "endpoint", headerByHeightEndpoint, "err", err)
+		return
+	}
+	_, err = w.Write(resp)
+	if err != nil {
+		log.Errorw("writing response", "endpoint", headerByHeightEndpoint, "err", err)
+		return
+	}
+}
+
+func (s *Service) handleRootRequest(w http.ResponseWriter, r *http.Request) {
```

Comment:
Forgot to rename?

Reply:
Fixed in [a46f9f9](https://github.com/celestiaorg/celestia-node/pull/672/commits/a46f9f982804131a8180f5be250af7c4c147e7d0)